pkg/crc/pullsecret/pullsecret.go

```diff
@@ -0,0 +1,173 @@
+		return err
+	}
+
+	// Replace user pull secret and add cluster ID
+	if err := setPullSecretAndClusterID(driver); err != nil {
+		return err
+	}
+
+	// Add user pull secret to the instance
+	if err := addpullSecretToIntanceDisk(driver, pullSec); err != nil {
+		return err
+	}
+	return nil
+}
+
+func addpullSecretSpecToInstance(driver drivers.Driver, pullSec string) error {
+	base64OfPullSec := base64.StdEncoding.EncodeToString([]byte(pullSec))
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/pull-secret.yaml\n%s\nEOF", fmt.Sprintf(pullSecret, base64OfPullSec)))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addpullSecretToIntanceDisk(driver drivers.Driver, pullSec string) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | sudo tee /var/lib/kubelet/config.json\n%s\nEOF", pullSec))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output is : %s", output)
+	return nil
+}
+
+func addKubeconfigFileToInstance(driver drivers.Driver, kubeconfigFilePath string) error {
+	kubeconfig, err := ioutil.ReadFile(kubeconfigFilePath)
+	if err != nil {
+		return err
+	}
+	_, err = drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf("cat <<EOF | tee /tmp/kubeconfig\n%s\nEOF", string(kubeconfig)))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func replaceUserPullSecret(driver drivers.Driver) error {
+	output, err := drivers.RunSSHCommandFromDriver(driver, replacePullSecretCmd)
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", replacePullSecretCmd, output)
+	return nil
+}
+
+func addClusterID(driver drivers.Driver) error {
+	clusterID := uuid.New()
+	output, err := drivers.RunSSHCommandFromDriver(driver, fmt.Sprintf(updateClusterIdCmd, clusterID))
+	if err != nil {
+		return err
+	}
+	logging.DebugF("Output of %s: %s", updateClusterIdCmd, output)
+	return nil
+}
+
+func setPullSecretAndClusterID(driver drivers.Driver) (rerr error) {
+	m := errors.MultiError{}
+	sd := systemd.NewInstanceSystemdCommander(driver)
+	defer func() {
+		// Stop the kubelet service.
+		if _, err := sd.Stop("kubelet"); err != nil {
+			m.Collect(err)
+		}
+		output, err := drivers.RunSSHCommandFromDriver(driver, stopAndRemovePodsCmd)
+		logging.DebugF("Output of %s: %s", stopAndRemovePodsCmd, output)
+		if err != nil {
+			m.Collect(err)
+		}
+		rerr = m.ToError()
+	}()
+
+	// Start the kubelet service
+	if _, err := sd.Start("kubelet"); err != nil {
+		m.Collect(err)
+	}
+
+	// Replace existing pull secret with the user pull secret
+	if err := replaceUserPullSecret(driver); err != nil {
+		m.Collect(err)
+	}
+
+	// add random cluster id
+	if err := addClusterID(driver); err != nil {
+		m.Collect(err)
+	}
+
+	return nil
+}
+
+func StorePullSecretToFile(pullsecret string, destDir string, fileName string, mode os.FileMode) error {
+	err := os.MkdirAll(destDir, mode)
```

Comment:
I'd hardcode 0700 here too  if you hardcode the 0600 below.

Reply:
@cfergeau no, all other directory are 0755 so I didn't hard code it.

Reply:
Use 0755 if you think it's better, but I would not add a `mode` argument.